File: Java/Easy/SearchInsertPos.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int out = 0;

        if (nums.length == 1){
            if (nums[0] > target || nums[0] == target){
                out = 0;
            } else {
                out = 1;
            }
        } else {
            for (int i = 0; i <= nums.length; i++){
                if (nums[i] == target){
                    return i;
                } else if (i == nums.length - 1){
                    out = nums.length;
                    break;
                } else if (nums[i] < target && nums[i+1] > target){
                    out = i+1;
                    break;
                } else if (nums[i] > target && i == 0){
                    out = 0;
                    break;
                }
            }
        }
        return out;
    }
}

File: Python/Easy/SearchInsertPos.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        out = 0
        if len(nums) == 1:
            if nums[0] > target or nums[0] == target:
                return 0
            else:
                return 1
        else:
            for i in range(len(nums)):
                if nums[i] == target:
                    return i
                elif (i == len(nums)-1):
                    return len(nums)
                    break
                elif nums[i] < target and nums[i+1] > target:
                    return i+1
                    break
                elif nums[i] > target and i == 0:
                    return 0
                    break
